### PR TITLE
Sync labels with istio/istio

### LINKED ISSUE #18773: `istioctl x describe` outputs `no ingress gateway service`
`istioctl x describe` is using an no-longer-supported selector to find Ingress pods.

cc @linsun 

See the last line of this output:

```
$ istioctl x describe pod istio-ingressgateway-6955467696-bxfdn -n istio-system
Pod: istio-ingressgateway-6955467696-bxfdn
   Pod Ports: 15020 (istio-proxy), 80 (istio-proxy), 443 (istio-proxy), 15029 (istio-proxy), 15030 (istio-proxy), 15031 (istio-proxy), 15032 (istio-proxy), 15443 (istio-proxy), 15011 (istio-proxy), 8060 (istio-proxy), 853 (istio-proxy), 15090 (istio-proxy)
Suggestion: add 'version' label to pod for Istio telemetry.
--------------------
Service: istio-ingressgateway
   Port: status-port 15020/UnsupportedProtocol targets pod port 15020
   Port: http2 80/HTTP2 targets pod port 80
   Port: https 443/HTTPS targets pod port 443
   Port: kiali 15029/UnsupportedProtocol targets pod port 15029
   Port: prometheus 15030/UnsupportedProtocol targets pod port 15030
   Port: grafana 15031/UnsupportedProtocol targets pod port 15031
   Port: tracing 15032/UnsupportedProtocol targets pod port 15032
   Port: tls 15443/TLS targets pod port 15443
15020 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15020 Pod is PERMISSIVE and clients are DISABLE
80 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
80 Pod is PERMISSIVE and clients are DISABLE
443 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
443 Pod is PERMISSIVE and clients are DISABLE
15029 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15029 Pod is PERMISSIVE and clients are DISABLE
15030 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15030 Pod is PERMISSIVE and clients are DISABLE
15031 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15031 Pod is PERMISSIVE and clients are DISABLE
15032 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15032 Pod is PERMISSIVE and clients are DISABLE
15443 DestinationRule: default for "*.local"
   Traffic Policy TLS Mode: DISABLE
15443 Pod is PERMISSIVE and clients are DISABLE
Error: no ingress gateway service
```

### PR DESCRIPTION
**Pull request by howardjohn**
I don't fully understand the details of it, but Kubernetes gets mad when
you change labels sometimes, our tools depend on some labels (like
istioctl), and users may have tools/scripts that do as well. For
consistency, make these the same as istio/istio

Fixes https://github.com/istio/istio/issues/18773

### CODE REVIEW

File: istio-policy/templates/poddisruptionbudget.yaml [hunk 1]
```diff
@@ -5,7 +5,7 @@ metadata:
   name: istio-policy
   namespace: {{ .Release.Namespace }}
   labels:
-    app: istio-policy
+    app: policy
```
Comment (ostromart) [inline istio-policy/templates/poddisruptionbudget.yaml]: should this be mixer?

File: istio-telemetry/mixer-telemetry/templates/poddisruptionbudget.yaml [hunk 2]
```diff
@@ -13,7 +13,7 @@ spec:
   minAvailable: 1
   selector:
     matchLabels:
-      app: istio-telemetry
+      app: telemetry
```
Comment (ostromart) [inline istio-telemetry/mixer-telemetry/templates/poddisruptionbudget.yaml]: mixer?

**Comment by ostromart** [comment]
If you get the right patch for gencheck could you apply it to my pr also? :-D
https://github.com/istio/installer/pull/524

File: istio-telemetry/mixer-telemetry/templates/poddisruptionbudget.yaml (hunk 2 above)
Comment (howardjohn) [reply istio-telemetry/mixer-telemetry/templates/poddisruptionbudget.yaml]: Oof good catch, I was looking at the wrong place

**Comment by howardjohn** [comment]
@ostromart I think I fixed everything

**Comment by howardjohn** [comment]
@istio/wg-environments-maintainers anyone around? This is a must have for 1.4 and we are hoping to build that ASAP

**Review by morvencao — APPROVED** [review]
/lgtm

**Pull request merged**